crates/transpiler/src/passes/consolidate_blocks.rs

```diff
@@ -349,7 +352,42 @@ pub fn run_consolidate_blocks(
     Ok(())
 }
 
+/// Replaces each block of consecutive gates by a single unitary node.
+///
+/// This is the main function of the `ConsolidateBlocks` transpiler pass
+/// which replaces uninterrupted sequences of gates acting on the same qubits
+/// into a Unitary node, which will consecutively be resynthesized into a
+/// more optimal subcircuit.
+///
+/// # Arguments
+/// * `dag` - The circuit for which we will consolidate gates.
+/// * `decomposer` - The type of decomposer to be used, could be of types
+///   [TwoQubitBasisDecomposer] or [TwoQubitControlledUDecomposer].
+/// * `basis_gate_name` - The basis gate selected for the decomposition.
+/// * `force_consolidate` - Decides whether to force all consolidations or not.
+/// * `target` - The target representing the backend for which the pass is consolidating.
+/// * `basis_gates` - The basis gates from which the decomposer will choose.
+pub fn run_consolidate_blocks(
+    dag: &mut DAGCircuit,
+    decomposer: DecomposerType,
```

Comment:
This is not something we should need as an input, it's something that can be detected very quickly from the `Target`. But I know you have a PR for this already.